### PR TITLE
More potential analysis tools

### PR DESCRIPTION
**Pull request by cortner**
Starting to move some of the potential analysis tools I've been using into ACE1pack. 

- [x] rdf
- [x] dimer curves
- [x] trimer curves
- [x] decohesion curves 
- [x] Also TODO: fix CI failure on nightly. This is related to method overwriting and happens to be methods related to *this* PR: 

```
ERROR: LoadError: Method overwriting is not permitted during Module precompile.
```
Easy fix, that would fit well into #144

### CODE REVIEW

**Comment by cortner** [comment]
@CheukHinHoJerry -- would you be willing to add the trimer curve and plts to this PR? Can be a rough draft for now. I suspect we will iterate quite a bit on how to best make these analysis tools available to users.

**Comment by CheukHinHoJerry** [comment]
Yes sure - will do this later tonight.

**Comment by CheukHinHoJerry** [comment]
I have no permission to push to this repo directly. Could you please take a look?

Update: Thank you. Just pushed.

**Comment by CheukHinHoJerry** [comment]
An equivalent function of :
```julia
function dimers(potential, elements; 
                rr = range(1e-3, _cutoff(potential), length=200), 
                minE = -1e10, maxE = 1e10, )
   zz = AtomicNumber.(elements)
   dimers = Dict() 
   for i = 1:length(zz), j = 1:i
      z1 = zz[i]
      z0 = zz[j]
      v01 = dimer_energy.(Ref(potential), rr, z1, z0)
      v01 = max.(min.(v01, maxE), minE)
      dimers[(z1, z0)] = (rr, v01)
   end
   return dimers
end
```
would be returning a Dict() swapping the range of given $\theta$, $r_1$ and $r_2$ and different combination of $z$s and that would be a huge Dictionary. Is that fine or I should make $r_1$ and $r_2$ as an arg?

**Comment by cortner** [comment]
this is ready I think, though the smoothness prior tutorial needs another edit by James (separate PR I think) and the new functions ought to be documented ...

File: ext/ACE1pack_Plots_ext.jl
Comment (wcwitt) [inline ext/ACE1pack_Plots_ext.jl]: @cortner, Can you make a final decision about this before we merge?

File: src/analysis/dataset_analysis.jl
Comment (wcwitt) [inline src/analysis/dataset_analysis.jl]: Any reason to keep this?

File: src/analysis/dataset_analysis.jl
Comment (wcwitt) [inline src/analysis/dataset_analysis.jl]: Any reason to keep this?

File: src/analysis/potential_analysis.jl
Comment (wcwitt) [inline src/analysis/potential_analysis.jl]: Should we remove this before the merge?

**Review by wcwitt — CHANGES_REQUESTED** [review]

File: ext/ACE1pack_Plots_ext.jl
Comment (cortner) [reply ext/ACE1pack_Plots_ext.jl]: I will remove this.

Comment (cortner) [reply ext/ACE1pack_Plots_ext.jl]: done

File: src/analysis/dataset_analysis.jl
Comment (cortner) [reply src/analysis/dataset_analysis.jl]: removed

File: src/analysis/dataset_analysis.jl
Comment (cortner) [reply src/analysis/dataset_analysis.jl]: removed

File: src/analysis/potential_analysis.jl
Comment (cortner) [reply src/analysis/potential_analysis.jl]: I wasn't sure which variant of the trimer function to use ... but this might be easy enough to recover with a few minutes' work. so yes I'll get rid of it too.

**Comment by cortner** [comment]
@wcwitt  -- can you please take one more look? I hope this is ready now.

**Comment by wcwitt** [comment]
brilliant, thanks

**Comment by cortner** [comment]
Thanks!

Before we merge your next PR can you please take a close look at the current main and what Ive done to achieve similar things?

If it's all clear to you how to proceed then please just ping me and explain. I'll then review.

**Pull request merged**
